### PR TITLE
Updates for .NET February 2024 Servicing

### PR DESCRIPTION
**Pull request by lbussell**
What's still left to do:

- [x] .NET Monitor Shas not yet available
- [x] .NET 9 Preview 1 Build is not yet finalized

### CODE REVIEW

File: manifest.versions.json [hunk 1]
```diff
@@ -87,131 +113,162 @@
...
+    "powershell|7.0|Windows|x64|sha": "6daac8fd09f1cea041a0b5afcbec0e9fc1b6b3dc38943068a0f1a76768882cd7a8876a3df9864cc7b0848a9d128f5cac93dd34f1522258fbc05e2133bb225603",
+
+    "powershell|8.0|build-version": "7.4.1",
+    "powershell|8.0|Linux.Alpine|sha": "c9e173a29e72e80e047b7bf719ed2796d4250ac534e7af833580f0193ff804896a327e51c5fa1f4aa3635fe5b3ce6bdec2ec7eb536f25e3650b36ea5591afa40",
+    "powershell|8.0|Linux|arm32|sha": "b4803878da4a5330942b0e4041bf7b830e658bd2450626712b0acfc9610212d1b3fcaacfc8fd53a27c8ba8e5225ac84dfe676a2afc182a040014f11835c841fb",
+    "powershell|8.0|Linux|arm64|sha": "30235cd50948a1020c0a4d33a8f8bc5ee37ef4246dc3f4dc7c1cf250645352af8d752cf823242e0917b6d3ab2ea307740fdd18600195a696b2343bc9b13005d4",
+    "powershell|8.0|Linux|x64|sha": "36a447aa79d408bf84320fd262e241770c9bb681710036e4a151a971418fb2b190e79e785b06b24e47651de53e4db252e80901f6975d08a14e6b7d59b59307e7",
+    "powershell|8.0|Windows|x64|sha": "d29f6f2f9e650d6fdb716c82b55396933122fe4706d833f89cd6d8a285325af6dc523930f828d1e0252d6c07aa341807b302e412cd268cb175cf1b5db9ccef44",
+
+    "powershell|9.0|build-version": "7.4.0-rc.1",
```
Comment (mthalman) [inline manifest.versions.json:165]: This should be `7.4.0`.

File: manifest.json [hunk 2]
```diff
@@ -1566,743 +1565,619 @@
...
           ]
         },
         {
-          "id": "bullseye-slim",
-          "productVersion": "$(dotnet|6.0|product-version)",
+          "productVersion": "$(dotnet|9.0|product-version)",
           "sharedTags": {
-            "$(dotnet|6.0|product-version)-bullseye-slim": {},
-            "6.0-bullseye-slim": {}
+            "$(dotnet|9.0|fixed-tag)-alpine3.18": {},
```
Comment (mthalman) [inline manifest.json:1618]: 9.0 should use 3.19, not 3.18.

File: tests/Microsoft.DotNet.Docker.Tests/TestData.cs [hunk 3]
```diff
@@ -148,11 +148,83 @@ public static class TestData
...
+            new ProductImageData { Version = V9_0, OS = OS.Jammy,               Arch = Arch.Amd64 },
+            new ProductImageData { Version = V9_0, OS = OS.JammyChiseled,       Arch = Arch.Amd64,   SdkOS = OS.Jammy },
+            new ProductImageData { Version = V9_0, OS = OS.JammyChiseled,       Arch = Arch.Amd64,   SdkOS = OS.Jammy },
+            new ProductImageData { Version = V9_0, OS = OS.JammyChiseled,       Arch = Arch.Amd64,   SdkOS = OS.Jammy,
+                    ImageVariant = DotNetImageVariant.Composite, SupportedImageRepos = DotNetImageRepo.Aspnet },
+            new ProductImageData { Version = V9_0, OS = OS.JammyChiseled,       Arch = Arch.Amd64,   SdkOS = OS.Jammy,
+                    ImageVariant = DotNetImageVariant.Extra, SupportedImageRepos = DotNetImageRepo.Runtime_Deps },
+            new ProductImageData { Version = V9_0, OS = OS.JammyChiseled,       Arch = Arch.Amd64,   SdkOS = OS.Jammy, SdkImageVariant = DotNetImageVariant.AOT,
+                    ImageVariant = DotNetImageVariant.AOT, SupportedImageRepos = DotNetImageRepo.Runtime_Deps },
+            new ProductImageData { Version = V9_0, OS = OS.Alpine318,           Arch = Arch.Amd64 },
```
Comment (mthalman) [inline tests/Microsoft.DotNet.Docker.Tests/TestData.cs:169]: Should be Alpine319

File: tests/Microsoft.DotNet.Docker.Tests/TestData.cs [hunk 4]
```diff
@@ -148,11 +148,83 @@ public static class TestData
                     ImageVariant = DotNetImageVariant.Composite, SupportedImageRepos = DotNetImageRepo.Aspnet },
             new ProductImageData { Version = V8_0, OS = OS.Alpine318,           Arch = Arch.Arm,     SdkOS = OS.Alpine318,
                     ImageVariant = DotNetImageVariant.Extra, SupportedImageRepos = DotNetImageRepo.Runtime_Deps },
+            new ProductImageData { Version = V8_0, OS = OS.Alpine318,           Arch = Arch.Arm,     SdkOS = OS.Alpine318, SdkImageVariant = DotNetImageVariant.AOT,
```
Comment (mthalman) [inline tests/Microsoft.DotNet.Docker.Tests/TestData.cs:151]: There are several references to AOT in the changes for this file. What is the impact of that if there are no AOT images defined?

**Review by jander-msft — COMMENTED** [review]
.NET Monitor 9.0 Preview 1 images are missing

File: README.runtime.md [hunk 5]
```diff
@@ -61,7 +61,7 @@ The [Image Variants documentation](https://github.com/dotnet/dotnet-docker/blob/
 ## Linux amd64 Tags
 Tags | Dockerfile | OS Version
 -----------| -------------| -------------
-8.0.2-bookworm-slim-amd64, 8.0-bookworm-slim-amd64, 8.0.2-bookworm-slim, 8.0-bookworm-slim, 8.0.2, 8.0 | [Dockerfile](https://github.com/dotnet/dotnet-docker/blob/main/src/runtime/8.0/bookworm-slim/amd64/Dockerfile) | Debian 12
+8.0.2-bookworm-slim-amd64, 8.0-bookworm-slim-amd64, 8.0.2, 8.0.2-bookworm-slim, 8.0, 8.0-bookworm-slim, latest | [Dockerfile](https://github.com/dotnet/dotnet-docker/blob/main/src/runtime/8.0/bookworm-slim/amd64/Dockerfile) | Debian 12
```
Comment (mthalman) [inline README.runtime.md:64]: The ordering isn't right here. It should be from most-specific to least, starting with all the concrete tags and then the floating tags. I notice this is wrong in other places too. It looks like at least some of this mis-ordering already exists in main.

```suggestion
8.0.2-bookworm-slim-amd64, 8.0-bookworm-slim-amd64, 8.0.2-bookworm-slim, 8.0-bookworm-slim, 8.0.2, 8.0, latest | [Dockerfile](https://github.com/dotnet/dotnet-docker/blob/main/src/runtime/8.0/bookworm-slim/amd64/Dockerfile) | Debian 12
```

**Comment by jander-msft** [comment]
> * [ ]  .NET Monitor Shas not yet available

.NET Monitor checksums are now available

File: manifest.versions.json [hunk 6]
```diff
@@ -87,131 +113,174 @@
     "mingit|x64|url": "https://github.com/git-for-windows/git/releases/download/v2.43.0.windows.1/MinGit-2.43.0-64-bit.zip",
     "mingit|x64|sha": "1905d93068e986258fafc69517df8fddff829bb2a289c1fa4dcc6cdf720ddf36",
 
-    "monitor|6.3|build-version": "6.3.4",
-    "monitor|6.3|linux-musl|x64|sha": "e58df34000772015e30ac161f0056cb0c9c2a7c98d176b157c92eeb7e80c219ce9df839a471e00c3baa23ac36968300fe59288a9ec8500363229b6b7923be1c9",
-    "monitor|6.3|linux-musl|arm64|sha": "3a0e735bd652640fab12a06712b34351e927bf8781976b2e895fbc2e629e3a39ebc9da9b00e1f1f9f7d6c86d222b88a8f23e6806339c0d08e037f407e11cc75a",
-    "monitor|6.3|linux|x64|sha": "6df820099d13c2f367682c6f6f5a635da2d0943b5d6760c7987521f8b62ac1339bd4e84144fee2a70b1c721c7c124b3e33f91342523a2e9610a492263b43e9ec",
-    "monitor|6.3|linux|arm64|sha": "d070bbe993eb325a8d52d23c14d4e7365eaa13f1aa023d5fa7b9db2630b16186a10f75091c0ab9c79109811c1cb18d36ede247cbd88bd698ebea75beb0b2e78a",
-    "monitor|6.3|product-version": "6.3.4",
+    "monitor|6.3|build-version": "6.3.5-servicing.24080.12",
```
Comment (jander-msft) [inline manifest.versions.json:116]: This needs to be `6.3.5` (without the label). Please run `.\Set-DotnetVersions.ps1 6.3 -MonitorVersion 6.3.5`

File: manifest.versions.json [hunk 7]
```diff
@@ -87,131 +113,174 @@
...
+    "monitor|6.3|linux|arm64|sha": "7fabe8e143fbd85757a7f3af10e51ec65c2afe927b9e4fd04161775740fe9060c862fcb71ecd04c859da331c8215a9f1d7e345f5d5f07aee65866e9c0ec9e61a",
+    "monitor|6.3|product-version": "6.3.5",
 
-    "monitor|7.3|build-version": "7.3.2",
-    "monitor|7.3|linux-musl|x64|sha": "591cdbe72e724ddd67a9a223df167e33a7f84e47caa33bcadad6c44cb44f54ccab3be379c48e02898c89e51e4645d2872eccf5c8da110f7798722c9451ff9270",
-    "monitor|7.3|linux-musl|arm64|sha": "8bbb3313e86a83d515a284878e1db3ee1d0df6b17afe49c263316e01aa134995d90205bd0db4be22d937a74eaee1aafe92ccd5a5bafcf8002f7fed46025366fc",
-    "monitor|7.3|linux|x64|sha": "9e1da6b41119874659e00cfae5c22e541041093ff1c55ecb2234610502e714868474c3bc52de78a650201d36271296069e8ed5dc8b3a4972954bffadd4478248",
-    "monitor|7.3|linux|arm64|sha": "9866bc99b71880f1ea305361e23703f34238546589b70e38b387000e141335908f0ebd2708f851ad911dc89d53460b98e6a9215a6c391e78b558cb55569de3c3",
-    "monitor|7.3|product-version": "7.3.2",
+    "monitor|7.3|build-version": "7.3.3-servicing.24080.13",
```
Comment (jander-msft) [inline manifest.versions.json:123]: This needs to be `7.3.3` (without the label). Please run `.\Set-DotnetVersions.ps1 7.3 -MonitorVersion 7.3.3`

File: manifest.versions.json [hunk 8]
```diff
@@ -87,131 +113,174 @@
...
+    "monitor|7.3|linux-musl|x64|sha": "c6ccbaa680b9aace18abf997437cf34dd0ea6eb74f6ec75bb346f1ff35a37c334db8cd8856e6cc4499398a064421d3d250a5d8fe68ccb5dd16039321bbe14140",
+    "monitor|7.3|linux-musl|arm64|sha": "e042cb72e70909052508bdedfc330481f68fd30e1c98d96591813ad4d77d1e8ce6ebcc6b2e172e0379b8fec8fb566661f37e4c18a6036a20be3ea95bd6536264",
+    "monitor|7.3|linux|x64|sha": "95b023caf844ad76bbba2917c0d620b87fe48b3951d3139943ecdd117fc6db3c470f9189ae30a229f248b385f5c4f7b56eabbc5998e656d01972956f4b7a8804",
+    "monitor|7.3|linux|arm64|sha": "3eb5fc7cecf95a4e82669f0cb7691a22d18787f2f9992f902e22cdd9318821cb2db2fd792dd15fe9baadc76c9fca9d414a3a6e0f4e3fc73e8ecd8c18f2ae3925",
+    "monitor|7.3|product-version": "7.3.3",
 
-    "monitor|8.0|build-version": "8.0.0",
-    "monitor|8.0|product-version": "8.0.0",
+    "monitor|8.0|build-version": "8.0.1-servicing.24081.5",
```
Comment (jander-msft) [inline manifest.versions.json:130]: This needs to be `8.0.1` (without the label). Please run `.\Set-DotnetVersions.ps1 8.0 -MonitorVersion 8.0.1`.

File: manifest.versions.json [hunk 9]
```diff
@@ -87,131 +113,174 @@
...
+    "monitor|8.0|product-version": "8.0.1",
 
-    "monitor-base|8.0|linux|x64|sha": "29e4b3d1d85dcc170ed2c6e6525345e4ba896ff525258ec7fb93b7b6980d048c859f8b6b892bbaacb19f0f8eda1616a118be8f6f50cec23a2aea67ba3cab3f31",
-    "monitor-base|8.0|linux|arm64|sha": "0acc71ab36c98b03c1c66589ac8a0656dee97a29f10b44a62d715936cdb003bfb5d754f6be6f65cb46fb3fa4066a93f45d88790aab0db1631714d27f56eabbce",
+    "monitor|9.0|build-version": "9.0.0-preview.1.24080.10",
```
Comment (jander-msft) [inline manifest.versions.json:133]: This version is not correct. Please run `.\Set-DotnetVersions.ps1 9.0 -MonitorVersion 9.0.0-preview.1.24106.4 `

**Review by jander-msft — APPROVED** [review]
.NET Monitor LGTM

**Review by mthalman — APPROVED** [review]

**Pull request merged**